test/robot/job/worker/worker.go

```diff
@@ -107,6 +107,22 @@ func RetryFunction(ctx context.Context, maxAttempts int, delay time.Duration, ta
 	return log.Errf(ctx, err, "Retried maximum attempts, %v", maxAttempts)
 }
 
+func WithTimeout(ctx context.Context, timeout time.Duration, task func() error) error {
+	done := make(chan error)
+	go func() {
+		done <- task()
+	}()
+	t := make(chan bool)
+	timer := time.AfterFunc(timeout, func() { t <- true })
+	defer timer.Stop()
+	select {
+	case err := <-done:
+		return err
+	case <-t:
```

Comment:
`case <-time.After(timeout):`

Reply:
This does not have a `stop()` function, so if the task returns first, the `time.After` will be leaked.